### PR TITLE
Expose Attachments in Mailbox Preview

### PR DESCRIPTION
**Pull request by maennchen**
screen shot 2018-06-05 at 17 23 48

### CODE REVIEW

File: lib/plug/mailbox_preview.ex [hunk 1]
```diff
@@ -53,6 +53,35 @@ if Code.ensure_loaded?(Plug) do
...
+    get "/:id/attachments/:index" do
+      index = String.to_integer(index)
+      id
+      |> conn.assigns.storage_driver.get()
+      |> case do
+        %{attachments: attachments} when length(attachments) > index ->
+          attachments
+          |> Enum.at(index)
+          |> case do
+            %{data: data, content_type: content_type} ->
```
Comment (princemaple) [inline lib/plug/mailbox_preview.ex]: This will probably always match? There is no nil guard here and the struct always has all the keys.

Comment (maennchen) [reply lib/plug/mailbox_preview.ex]: You’ probably right, I’m going to fix that.

Comment (maennchen) [reply lib/plug/mailbox_preview.ex]: Done

File: lib/plug/mailbox_preview.ex [hunk 2]
```diff
@@ -53,6 +53,35 @@ if Code.ensure_loaded?(Plug) do
...
+    get "/:id/attachments/:index" do
+      index = String.to_integer(index)
+      id
+      |> conn.assigns.storage_driver.get()
+      |> case do
+        %{attachments: attachments} when length(attachments) > index ->
+          attachments
+          |> Enum.at(index)
+          |> case do
+            %{data: data, content_type: content_type} when not is_nil(data) ->
```
Comment (maennchen) [inline lib/plug/mailbox_preview.ex]: Sorry about than one, I'm a little bit in a hurry today.

Comment (princemaple) [reply lib/plug/mailbox_preview.ex]: No worries!

**Comment by princemaple** [comment]
Thank you again for the PR!

**Comment by princemaple** [comment]
@maennchen released in v0.15

**Pull request merged**